In [1]:
import pandas as pd
import math
from scipy.stats import ttest_ind

# Exercise 1

In [2]:
data1 = pd.read_csv('files_for_lab/machine.txt', sep='\t', header=0, names=['new_machine', 'old_machine'])
data1.columns = [columns_name.lower() for columns_name in data1.columns]
data1.columns = [columns_name.replace(' ', '_') for columns_name in data1.columns]
data1

new_machine  old_machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

In [3]:
# new_machine
nm_samples = len(data1['new_machine'])
nm_sample_mean = data1['new_machine'].mean()
nm_sample_std = data1['new_machine'].std()

# old_machine
om_samples = len(data1['old_machine'])
om_sample_mean = data1['old_machine'].mean()
om_sample_std = data1['old_machine'].std()

print('number of records:->', 'new machine:', nm_samples, '/', 'old machine:', om_samples)
print()
print('new machine packing time->', 'mean:', nm_sample_mean, '&', 'standard deviation:', nm_sample_std)
print()
print('old machine packing time->', 'mean:', om_sample_mean, '&', 'standard deviation:', om_sample_std)

number of records:-> new machine: 10 / old machine: 10

new machine packing time-> mean: 42.14 & standard deviation: 0.6834552736727638

old machine packing time-> mean: 43.230000000000004 & standard deviation: 0.7498888806572157


In [4]:
new_machine = data1['new_machine'].to_numpy()
old_machine = data1['old_machine'].to_numpy()

print(new_machine)
print()
print(old_machine)

[42.1 41.  41.3 41.8 42.4 42.8 43.2 42.3 41.8 42.7]

[42.7 43.6 43.8 43.3 42.5 43.5 43.1 41.7 44.  44.1]


In [5]:
ttest_ind(new_machine, old_machine)

Ttest_indResult(statistic=-3.3972307061176026, pvalue=0.0032111425007745158)

In this lab, the null hypothesis is that there is no significant difference between the mean packing times of the new and old machines, while the alternative hypothesis is that the mean packing time of the new machine is greater than the mean packing time of the old machine.

### Conclusions 

Based on the t-test result, we can conclude that the p-value is less than the significance level (alpha) typically used in hypothesis testing, such as 0.05 or 0.01. In this case, the p-value is 0.003, which is less than 0.05, so we can reject the null hypothesis and conclude that there is a significant difference between the mean packing times of the new and old machines.

The negative value of the t-statistic (-3.397) indicates that the mean packing time of the old machine is higher than the mean packing time of the new machine. Therefore, we can conclude that there is evidence that the new machine packs faster on average than the old machine.

# Exercise 2

In [6]:
data2 = pd.read_csv('files_for_lab/student_gpa.txt', sep='\t', header=0, names=['Sophomores', 'Juniors'])
data2.columns = [columns_name.lower() for columns_name in data2.columns]
data2.columns = [columns_name.replace(' ', '_') for columns_name in data2.columns]
data2

sophomores  juniors
0         3.04     2.56
1         1.71     2.77
2         3.30     2.70
3         2.88     3.00
4         2.11     2.98
5         2.60     3.47
6         2.92     3.26
7         3.60     3.20
8         2.28     3.19
9         2.82     2.65
10        3.03     3.00
11        3.13     3.39
12        2.86     2.58
13        3.49      NaN
14        3.11      NaN
15        2.13      NaN
16        3.27      NaN

In [7]:
# sophomores
nm_samples_s = len(data2['sophomores'])
nm_sample_mean_s = data2['sophomores'].mean()
nm_sample_std_s = data2['sophomores'].std()

# juniors
om_samples_j = len(data2['juniors'])
om_sample_mean_j = data2['juniors'].mean()
om_sample_std_j = data2['juniors'].std()

print('number of records:->', 'sophomores:', nm_samples_s, '/', 'old machine:', om_samples_j)
print()
print('sophomores scores->', 'mean:', nm_sample_mean_s, '&', 'standard deviation:', nm_sample_std_s)
print()
print('juniors scores->', 'mean:', om_sample_mean_j, '&', 'standard deviation:', om_sample_std_j)

number of records:-> sophomores: 17 / old machine: 17

sophomores scores-> mean: 2.8400000000000003 & standard deviation: 0.5198317035349037

juniors scores-> mean: 2.980769230769231 & standard deviation: 0.3092588327615327


In [8]:
import numpy as np

sophomores = data2['sophomores'].to_numpy()
juniors = data2['juniors'].to_numpy()
juniors = juniors[~np.isnan(juniors)]

print(sophomores)
print()
print(juniors)

[3.04 1.71 3.3  2.88 2.11 2.6  2.92 3.6  2.28 2.82 3.03 3.13 2.86 3.49
 3.11 2.13 3.27]

[2.56 2.77 2.7  3.   2.98 3.47 3.26 3.2  3.19 2.65 3.   3.39 2.58]


In [9]:
ttest_ind(sophomores, juniors, equal_var=False) # equal_var=False to specify that the variances are not equal since the lab states "In this case we can't assume that the population variances are equal"

Ttest_indResult(statistic=-0.9231495630900278, pvalue=0.3642180675348571)

The null hypothesis for this test is that there is no significant difference in the mean GPAs of sophomores and juniors at the university. The alternative hypothesis is that they are different.

### Conclusions

Since the p-value is not less than the significance level of 0.05, we do not have sufficient evidence to reject the null hypothesis that the mean GPAs of sophomores and juniors are equal (not significantly different). We cannot conclude that the GPAs from one group and the other are somewhat equal, but rather that there is insufficient evidence to suggest they differ.